presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java

```diff
@@ -26,21 +28,24 @@
 
 import static com.facebook.presto.hive.HiveErrorCode.HIVE_FILESYSTEM_ERROR;
 import static com.facebook.presto.hive.HiveErrorCode.HIVE_PATH_ALREADY_EXISTS;
-import static com.facebook.presto.hive.HiveWriteUtils.createTemporaryPath;
+import static com.facebook.presto.hive.HiveWriteUtils.createDirectory;
 import static com.facebook.presto.hive.HiveWriteUtils.getTableDefaultLocation;
 import static com.facebook.presto.hive.HiveWriteUtils.pathExists;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
 
 public class HiveLocationService
         implements LocationService
 {
     private final HiveMetastore metastore;
     private final HdfsEnvironment hdfsEnvironment;
+    private final HiveClientConfig hiveClientConfig;
 
     @Inject
-    public HiveLocationService(HiveMetastore metastore, HdfsEnvironment hdfsEnvironment)
+    public HiveLocationService(HiveMetastore metastore, HdfsEnvironment hdfsEnvironment, HiveClientConfig hiveClientConfig)
     {
+        this.hiveClientConfig = hiveClientConfig;
```

Comment:
Configuration objects are mutable and can not be held onto after the injection phase.  Instead copy out the values you are interested in.